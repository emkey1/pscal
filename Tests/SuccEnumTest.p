program EnumSuccTest;

type
  TColor = (cRed, cGreen, cBlue, cYellow);

procedure AssertEqualEnum(expected, actual: TColor; testName: string);
begin
  if ord(expected) = ord(actual) then
    writeln(testName, ': PASS')
  else
    writeln(testName, ': FAIL (expected ', ord(expected), ', got ', ord(actual), ')');
end;

begin
  AssertEqualEnum(cGreen, succ(cRed), 'Succ(cRed)');
  AssertEqualEnum(cYellow, succ(cBlue), 'Succ(cBlue)');
end.
