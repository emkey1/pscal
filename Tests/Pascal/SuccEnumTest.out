Succ(cRed): PASS
Succ(cGreen): PASS
Succ(cBlue): PASS
